pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -937,19 +951,7 @@ template <class Precision> class StateVectorKokkos {
     void
     applyPauliY(const std::vector<size_t> &wires, bool inverse = false,
                 [[maybe_unused]] const std::vector<Precision> &params = {}) {
-        auto &&num_qubits = getNumQubits();
-        assert(wires.size() == 1);
-        if (!inverse) {
-            Kokkos::parallel_for(Kokkos::RangePolicy<KokkosExecSpace>(
-                                     0, Util::exp2(num_qubits - 1)),
-                                 applyPauliYFunctor<Precision, false>(
-                                     *data_, num_qubits, wires));
-        } else {
-            Kokkos::parallel_for(
-                Kokkos::RangePolicy<KokkosExecSpace>(
-                    0, Util::exp2(num_qubits - 1)),
-                applyPauliYFunctor<Precision, true>(*data_, num_qubits, wires));
-        }
+        applySpecialQubitGate<applyPauliYFunctor, 1>(wires, inverse, params);
```

Comment:
> Since the applying action is living in the new `applySpecialQubitGate`, we could maybe rename all the family of `applySomethingFunctor`, to just `SomethingFunctor`. They are contained in `GateFunctors.hpp`. What do you think?

Agreed, I like shorter names too.

Reply:
> Also, `applySpecialQubitGate` could be called `applyGateFunctor`.

I feel like `applyGateFunctor` is a bit too general, and it does not encompasses all gates, but _special_ may be too vague. I was thinking of other adjectives: named, specific, common. Should we vote?

Reply:
What makes these gates special?

Reply:
It is just by opposition to _generic_ gates, which take an additional matrix argument.

Reply:
So, that's why I suggested `applyGateFunctor`, because it applies a gate defined by a functor.
If not I would have suggested just `applyGate`. :smile:
I think that when we define it in terms of a matrix we don't usually call it `applyGate`.
I'm going with what you think is better, but in case you proceed with Special (or something similar) it would be nice to provide a context for it in the documentation.

Reply:
Ah I see. I guess the generic matrix-one would be an _operator_ or something. I'll trust you on this one as I am still getting my quantum computation jargon right.

Reply:
I prefer `applyGateFunctor` because it is more explicit, and there could be other ways to apply gates (e.g. lambda functions).

Reply:
Yes! Perfect!